### DISCUSSION TITLE
Refactoring with autoimports

### DISCUSSION THREAD

**Discussion by YaredFall** [nuxt/nuxt#27682] [Questions]
When you trying to refactor (e.g. rename or move) autoimported componentents/composables/utils with your IDE, references of the refactored symbols not being updated.

For exapmle, if you have component `Foo.vue` in some other component `Bar.vue`.
```(vue)
<!-- Bar.vue before -->
<template>
    <Foo />
</template>
```
When you rename `Foo.vue` to `Foo1.vue`, the code inside the `Bar.vue` stays the same.

Expected output:
```(vue)
<!-- Bar.vue after -->
<template>
    <Foo1 />
</template>
```
Similar thing applies to any autoimported composable or util.

That quirk makes refactoring almost impossible in any mid to large codebase.

So, am I missing someting? How do you approach the refactoring in your Nuxt projects?

> P.S. I'm using `VSCode` with latest *(v2.0.21 at the moment)* `Vue - Offcial` extension

**Comment by michael-stubenvoll-heller** [comment]
I have the same problem. And I'm a little confused that I can't find anything written about whether this is a limitation when using auto imports and if I need to disable auto imports and import everything manually to enable refactoring support of my IDE. I thought refactoring is a very basic operation, which should be supported in a way.
